Crash - Mavericks IllegalArgumentException in QrCodeScannerFragment

```
java.lang.RuntimeException: 
  at android.app.ActivityThread.performResumeActivity (ActivityThread.java:4539)
  at android.app.ActivityThread.handleResumeActivity (ActivityThread.java:4575)
  at android.app.servertransaction.ResumeActivityItem.execute (ResumeActivityItem.java:52)
  at android.app.servertransaction.TransactionExecutor.executeLifecycleState (TransactionExecutor.java:181)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:102)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2235)
  at android.os.Handler.dispatchMessage (Handler.java:107)
  at android.os.Looper.loop (Looper.java:227)
  at android.app.ActivityThread.main (ActivityThread.java:7822)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:492)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1027)
Caused by: java.lang.IllegalArgumentException: 
  at com.airbnb.mvrx.MavericksExtensionsKt$args$1.getValue (MavericksExtensions.kt:8)
  at im.vector.app.features.qrcode.QrCodeScannerFragment.getScannerArgs (QrCodeScannerFragment.kt:1)
  at im.vector.app.features.qrcode.QrCodeScannerFragment.onViewCreated (QrCodeScannerFragment.kt:2)
  at androidx.fragment.app.Fragment.performViewCreated (Fragment.java:1)
  at androidx.fragment.app.FragmentStateManager.createView (FragmentStateManager.java:39)
  at androidx.fragment.app.FragmentStateManager.moveToExpectedState (FragmentStateManager.java:27)
  at androidx.fragment.app.FragmentManager.executeOpsTogether (FragmentManager.java:122)
  at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute (FragmentManager.java:9)
  at androidx.fragment.app.FragmentManager.execPendingActions (FragmentManager.java:14)
  at androidx.fragment.app.FragmentActivity.onResume (FragmentActivity.java:5)
  at im.vector.app.core.platform.VectorBaseActivity.onResume (VectorBaseActivity.kt:1)
  at android.app.Instrumentation.callActivityOnResume (Instrumentation.java:1453)
  at android.app.Activity.performResume (Activity.java:8125)
  at android.app.ActivityThread.performResumeActivity (ActivityThread.java:4529)
```

| Occurrences (30 days) | Total Occurrences | Impacted users (lifetime) |
| --- | --- | --- |
| 61 | 76 | 72 |

https://play.google.com/console/u/0/developers/{redacted}/app/{redacted}/vitals/crashes/99f978d5/details?versionCode=40104002/details?installedFrom=PLAY_STORE&days=30

--- a/changelog.d/5295.bugfix
+++ b/changelog.d/5295.bugfix
@@ -1,0 +1,1 @@
+Fixing crash when adding room by QR code after accepting the camera permission for the first time

--- a/vector/src/main/java/im/vector/app/features/createdirect/CreateDirectRoomActivity.kt
+++ b/vector/src/main/java/im/vector/app/features/createdirect/CreateDirectRoomActivity.kt
@@ -152,7 +152,8 @@ class CreateDirectRoomActivity : SimpleFragmentActivity() {
 
     private val permissionCameraLauncher = registerForPermissionsResult { allGranted, deniedPermanently ->
         if (allGranted) {
-            addFragment(views.container, QrCodeScannerFragment::class.java)
+            val args = QrScannerArgs(showExtraButtons = false, R.string.add_by_qr_code)
+            addFragment(views.container, QrCodeScannerFragment::class.java, args)
         } else if (deniedPermanently) {
             onPermissionDeniedSnackbar(R.string.permissions_denied_qr_code)
         }

--- a/vector/src/main/java/im/vector/app/features/qrcode/QrCodeScannerFragment.kt
+++ b/vector/src/main/java/im/vector/app/features/qrcode/QrCodeScannerFragment.kt
@@ -56,7 +56,7 @@ data class QrScannerArgs(
 open class QrCodeScannerFragment @Inject constructor() : VectorBaseFragment<FragmentQrCodeScannerBinding>(), ZXingScannerView.ResultHandler {
 
     private val qrViewModel: QrCodeScannerViewModel by activityViewModel()
-    private val scannerArgs: QrScannerArgs? by args()
+    private val scannerArgs: QrScannerArgs by args()
 
     override fun getBinding(inflater: LayoutInflater, container: ViewGroup?): FragmentQrCodeScannerBinding {
         return FragmentQrCodeScannerBinding.inflate(inflater, container, false)
@@ -93,13 +93,13 @@ open class QrCodeScannerFragment @Inject constructor() : VectorBaseFragment<Frag
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
-        val title = scannerArgs?.titleRes?.let { getString(it) }
+        val title = scannerArgs.titleRes.let { getString(it) }
 
         setupToolbar(views.qrScannerToolbar)
                 .setTitle(title)
                 .allowBack(useCross = true)
 
-        scannerArgs?.showExtraButtons?.let { showButtons ->
+        scannerArgs.showExtraButtons.let { showButtons ->
             views.userCodeMyCodeButton.isVisible = showButtons
             views.userCodeOpenGalleryButton.isVisible = showButtons
 

--- a/vector/src/main/java/im/vector/app/features/qrcode/QrCodeScannerFragment.kt
+++ b/vector/src/main/java/im/vector/app/features/qrcode/QrCodeScannerFragment.kt
@@ -53,7 +53,7 @@ data class QrScannerArgs(
         @StringRes val titleRes: Int
 ) : Parcelable
 
-open class QrCodeScannerFragment @Inject constructor() : VectorBaseFragment<FragmentQrCodeScannerBinding>(), ZXingScannerView.ResultHandler {
+class QrCodeScannerFragment @Inject constructor() : VectorBaseFragment<FragmentQrCodeScannerBinding>(), ZXingScannerView.ResultHandler {
 
     private val qrViewModel: QrCodeScannerViewModel by activityViewModel()
     private val scannerArgs: QrScannerArgs by args()

